core/modules/dblog/dblog.module

```diff
@@ -151,14 +151,19 @@ function dblog_watchdog(array $log_entry) {
  * Implements hook_form_FORM_ID_alter() for system_logging_settings().
  */
 function dblog_form_system_logging_settings_alter(&$form, $form_state) {
+  // Make sure the Date module file is available so we can call
+  // date_format_type_options().
+  if (!module_exists('date')) {
+    form_load_include($form_state, 'module', 'date');
```

Comment:
Agreed. We shouldn't be making dependencies between optional core modules.

Because `date_format_type_options()` calls `system_get_date_formats()` (in system.module), I think the most appropriate place would be making a new function in system module (e.g. `system_date_format_options()`) and then deprecating `date_format_type_options()`. Note that `date_example_date()` from date.module is also called, that can be moved to `date.inc`.

Reply:
FTR, I don't see this as a hack. I see this as using the `form_load_include()` function precisely for what it was created for.

Here's a similar, but different, example: https://api.backdropcms.org/api/backdrop/core%21modules%21locale%21locale.admin.inc/function/locale_date_format_form/1
At the top of that file you'll see it includes the `system.admin.inc` file in order to call `system_date_time_lookup()`.

But, if you're really set on removing this call to the core Date module, here's an example from Views which instead calls `system_get_date_formats()` and then generates the options array from that manually: https://api.backdropcms.org/api/backdrop/core%21modules%21views%21handlers%21views_handler_field_date.inc/function/views_handler_field_date%3A%3Aoptions_form/1
I guess we could do something similar here, even though it's more code...

Reply:
> I see this as using the form_load_include() function precisely for what it was created for.

This function wasn't created so that we could ignore a module's enabled state, it is intended to load files that might not be loaded due to a form cache. 

> At the top of that file you'll see it includes the system.admin.inc file in order to call system_date_time_lookup().

System is not an optional module, so this isn't an apples-to-apples comparison. 

> here's an example from Views...

If we move `date_format_type_options()` (or a comparable function with a different name) to a location where other modules could use it safely, we could continue to use it in `dblog`, and `views` could use it also. Not only do we fix our problem now, but we add a useful tool for other modules (even contrib!) to use when they need to present a list of date formats in an options form. :)